bin/lint-migrations-file/src/change_set/strict.clj

```diff
@@ -122,6 +122,30 @@
    (some change-types-supporting-rollback (mapcat keys changes))
    (contains? change-set :rollback)))
 
+(def ^:private change-types-requiring-preconditions
+  #{:createTable
+    :dropTable
+    :addColumn
+    :dropColumn
+    :createIndex
+    :dropIndex
+    :addForeignKeyConstraint
+    :dropForeignKeyConstraint})
+
+(defn- precondition-present-when-required?
+  "Ensures that certain changeSet types include a preCondition. The intent is for the preCondition to ensure that the changeSet is
+  idempotent by checking whether the table/column/index/etc does not exist before trying to create it. (Or inversely, that it does
+  exist before trying to drop it.)
+
+  We don't currently assert on the structure of the preCondition to provide flexibility if there are cases where idempotence is not
+  desired."
+  [{:keys [id changes] :as change-set}]
+  (or
+   (int? id)
```

Comment:
```suggestion
```

I don't think we need this anymore because the 001_update_migrations.yml only contains migration from v45->, so there are no int migrations

Reply:
Reverted this since a bunch of tests use int migration IDs. And it's simple enough to allow them.

If we want to explicitly lint against them that can be a separate PR, I want to keep this one focused.